selvpc/resource_selvpc_resell_crossregion_subnet_v2.go

```diff
@@ -0,0 +1,211 @@
+package selvpc
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/selectel/go-selvpcclient/selvpcclient/resell/v2/crossregionsubnets"
+)
+
+func resourceResellCrossRegionSubnetV2() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceResellCrossRegionSubnetV2Create,
+		Read:   resourceResellCrossRegionSubnetV2Read,
+		Delete: resourceResellCrossRegionSubnetV2Delete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"regions": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				Set:      hashRegions,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"region": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+			"cidr": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"vlan_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"servers": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Set:      hashServers,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"subnets": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Set:      hashSubnets,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"region": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"cidr": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"network_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"subnet_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"project_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"vlan_id": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"vtep_ip_address": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceResellCrossRegionSubnetV2Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	resellV2Client := config.resellV2Client()
+	ctx := context.Background()
+
+	projectID := d.Get("project_id").(string)
```

Comment:
We don't need `projectID` till L#138 and we can exit early in case of error in L#125.
So it's better to not initialise this variable that early.